### PR TITLE
fix(cmf/sagas): http response could not have headers

### PR DESCRIPTION
**Pull request by frassinier**
**What is the problem this PR is trying to solve?**
If we get a response without headers it will fails

**What is the chosen solution to this problem?**
Just check if http response has headers before dealing with them.

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/cmf/src/sagas/http.js [hunk 1]
```diff
@@ -49,11 +49,14 @@ export class HTTPError extends Error {
 export function handleBody(response) {
 	let methodBody = 'text';
 
-	const contentType = response.headers.get('Content-Type');
+	if (response.headers) {
```
Comment (sgendre) [inline packages/cmf/src/sagas/http.js]: get content-type with lodash get, provide a default value and avoid the unnecessary if statements

**Comment by ncomont** [comment]
@sgendre 

A _simple_ lodash `get` implies :

- lodash/get.js
2 ternary
1 call to `baseGet` function

- lodash/.internal/baseGet.js
1 call to `castPath` function
1 while loop based on the path's length
n calls to the `toKey` function
N ternaries

- lodash/.internal/castPath.js
1 call to `Array.isArray`
1 call to `isKey` function
1 call to the `stringToPath` function
1 ternary


- lodash/.internal/toKey.js
1 call to the `isSymbol` function
n ternaries
n arithmetical operations

etc, etc, etc.

I could continue again and again.

Why should we use complex libraries when our needs are so simple that a regular if could do the job ?

**Comment by acateland** [comment]
@ncomont the idea is to use the same pattern around access to unsafe value.

**Comment by sgendre** [comment]
the point is to avoid 'undefined is not a function' and also having a lot of if statements all along

my 2 cents
``export function handleBody(response) {
	if (get(response, 'headers', {}).get('Content-Type', []).includes('application/json')) {
		return response['json']().then(data => ({ data, response }));
	}

	return response['text']().then(data => ({ data, response }));
}``

File: packages/cmf/src/sagas/http.js [hunk 2]
```diff
@@ -2,6 +2,7 @@ import { call, put } from 'redux-saga/effects';
 import merge from 'lodash/merge';
 import get from 'lodash/get';
 import curry from 'lodash/curry';
+import { Headers } from 'node-fetch';
```
Comment (sgendre) [inline packages/cmf/src/sagas/http.js]: it's frontend code, what is the point of a node library ?

**Pull request merged**